machine_learning_hep/analysis/analyzer_jets.py

```diff
@@ -75,27 +82,28 @@ def __init__(self, datap, case, typean, period):
         self.h_eff_ptjet_pthf = {}
         self.h_effnew_ptjet_pthf = {'pr': None, 'np': None}
         self.h_effnew_pthf = {'pr': None, 'np': None}
-        self.hfeeddown_det = { 'mc': {}, 'data': {}}
+        self.hfeeddown_det = {'mc': {}, 'data': {}}
         self.h_reflcorr = create_hist('h_reflcorr', ';p_{T}^{HF} (GeV/#it{c})', self.bins_candpt)
         self.n_events = {}
         self.n_colls = {}
 
-        self.path_fig = Path(f'fig/{self.case}/{self.typean}')
+        self.path_fig = Path(f'{os.path.expandvars(self.d_resultsallpdata)}/fig')
         for folder in ['qa', 'fit', 'roofit', 'sideband', 'signalextr', 'sidesub', 'sigextr', 'fd', 'uf', 'eff']:
             (self.path_fig / folder).mkdir(parents=True, exist_ok=True)
 
-        self.rfigfile = TFile(str(self.path_fig / 'output.root'), 'recreate')
+        self.file_out_histo = TFile(self.n_fileresult, 'recreate')
 
         self.fitter = RooFitter()
         self.roo_ws = {}
         self.roo_ws_ptjet = {}
         self.roows = {}
 
+        self.n_iter_unfold_sel = datap["analysis"][typean]["unfolding_iterations_sel"]
+        self.n_rebin = datap["analysis"][typean]["n_rebin"]
```

Comment:
Why do we need these? Can't we use `self.cfg(...)` directly?

Reply:
Do you mean `self.n_rebin = self.cfg("n_rebin")` or replacing `self.n_rebin` completely?
In the latter case, I think the benefit of defining the variable is that the processing will crash already in `init` if the key does not exist in the database.`

Reply:
I mean using `self.cfg("n_rebin", 1)` in the code where it is used. This avoids the indirection through a member variable, which avoids having to first check how the member variable is defined. So, unless there is any additional logic in the definition of the member variable (like for some of the paths), I really don't see the point of having it at all.